core/services/src/main/java/org/exoplatform/services/attachments/service/AttachmentServiceImpl.java

```diff
@@ -282,6 +301,40 @@ public Attachment getAttachmentById(String attachmentId, SessionProvider session
     }
   }
 
+  @Override
+  public void moveAttachmentToNewPath(long userIdentityId,
+                                      String attachmentId,
+                                      String newPathDrive,
+                                      String newPath) throws Exception {
+    if (userIdentityId <= 0) {
+      throw new IllegalAccessException("User identity must be positive");
+    }
+    if (StringUtils.isEmpty(attachmentId)) {
+      throw new IllegalArgumentException("Attachment id is mandatory");
+    }
+    if (StringUtils.isEmpty(newPathDrive)) {
+      throw new IllegalArgumentException("New destination path's drive must not be empty");
+    }
+
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
```

Comment:
Please avoid using system session, rather use user session for permission checks

Reply:
okay